contracts/DelegateBorrowers/MoveDebtDelegate.sol

```diff
@@ -0,0 +1,161 @@
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { ResilientOracleInterface } from "@venusprotocol/oracle/contracts/interfaces/OracleInterface.sol";
+
+import { approveOrRevert } from "../lib/approveOrRevert.sol";
+import { IVBep20, IComptroller } from "../InterfacesV8.sol";
+
+contract MoveDebtDelegate is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable {
+    /// @dev VToken return value signalling about successful execution
+    uint256 internal constant NO_ERROR = 0;
+
+    /// @notice VToken to repay the debt to
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IVBep20 public immutable vTokenToRepay;
+
+    /// @notice Emitted if debt is swapped successfully
+    event DebtMoved(
+        address indexed originalBorrower,
+        address vTokenRepaid,
+        uint256 repaidAmount,
+        address indexed newBorrower,
+        address indexed vTokenBorrowed,
+        uint256 borrowedAmount
+    );
+
+    /// @notice Emitted when the owner transfers tokens, accidentially sent to this contract,
+    ///   to their account
+    event SweptTokens(address indexed token, uint256 amount);
+
+    /// @notice Thrown if VTokens' comptrollers are not equal
+    error ComptrollerMismatch();
+
+    /// @notice Thrown if repayment fails with an error code
+    error RepaymentFailed(uint256 errorCode);
+
+    /// @notice Thrown if borrow fails with an error code
+    error BorrowFailed(uint256 errorCode);
+
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    /// @notice Constructor for the implementation contract. Sets immutable variables.
+    /// @param vTokenToRepay_ VToken to repay the debt to
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(IVBep20 vTokenToRepay_) {
+        vTokenToRepay = vTokenToRepay_;
+        _disableInitializers();
+    }
+
+    function initialize() external initializer {
+        __Ownable2Step_init();
+        __ReentrancyGuard_init();
+    }
+
+    /**
+     * @notice Repays originalBorrower's borrow in vTokenToRepay.underlying() and borrows
+     *   vTokenToBorrow.underlying() on behalf of newBorrower
+     * @param originalBorrower The address of the borrower, whose debt to repay
+     * @param repayAmount The amount to repay in terms of vTokenToRepay.underlying()
+     * @param newBorrower The address of the borrower, who will own the new borrow
+     * @param vTokenToBorrow VToken to borrow from
+     */
+    function moveDebt(
+        address originalBorrower,
+        uint256 repayAmount,
+        address newBorrower,
+        IVBep20 vTokenToBorrow
```

Comment:
We could check this `vTokenToBorrow` is in one pre-fixed list of vtokens, like vUSDT and vUSDC

Reply:
[da6a48f](https://github.com/VenusProtocol/venus-protocol/pull/406/commits/da6a48f44e088bc73599c2abbe5a001409509d58)